buggy_data_structures.py

```
@@ -0,0 +1,116 @@
+        else:
+            raise ValueError(f"Unsupported format: {format_type}")
+    except (TypeError, ValueError) as e:
+        raise ValueError(f"Serialization failed: {e}")
+
+def deserialize_data(data, format_type):
+    # Bug fixed: add error handling for corrupted data
+    try:
+        if format_type == "json":
+            return json.loads(data)
+        elif format_type == "pickle":
+            return pickle.loads(data)
```

Ruff (0.12.2)

`pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)